Phonebook.Frontend/src/app/services/api/organigram.service.ts

```diff
@@ -110,7 +110,10 @@ export class OrganigramService {
     if (user.Business.ShortOrgUnit.length > depth) {
       for (const node of nodeChilds) {
         if (node.id === user.Business.ShortOrgUnit[depth] && node.depth === depth) {
-          return node;
+          if (user.Business.ShortOrgUnit.length == depth + 1) {
```

Comment:
Ich bin ja gegen backend spezifische Logik im Frontend, aber wenn du es so lassen willst schreib wenigsten ein Kommentar ran, warum plötzlich so etwas komisches passiert.

Reply:
Abgesehen davon glaube ich das diese Lösung woanders etwas kaputt macht.

Reply:
Die war vorher kaputt. Problem ist das in ShortOrgUnit ein Array ist welches von der obersten Ebene Anfängt und bis zur letzten Ebene geht. Also etwa so:

0 Firma
1 Bereich
2 Subbereich

Und der unterste Bereich ist der welcher auf den User Match bzw. In welcher der User sich befindet.

Somit weiß ich auch mit length == depth + 1 das ich mich auf dem letzten Sinnvollen such æren Ebene befinde und weiter unten in der Organisation nicht suchen muss.

Reply:
Aber müsste es dann nicht auch mit dem Code Vorher passen? 
Jetzt returned es ja einfach die jeweils tiefere Node ohne vorher zu gucken ob die id übereinstimmt?